core/src/test/java/org/apache/iceberg/TestFastAppend.java

```diff
@@ -478,4 +478,26 @@ public void testIncludedPartitionSummaryLimit() {
     String changedPartitions = table.currentSnapshot().summary().get(SnapshotSummary.CHANGED_PARTITION_COUNT_PROP);
     Assert.assertEquals("Should set changed partition count", "2", changedPartitions);
   }
+
+  @Test
+  public void testAppendToBranch() {
```

Comment:
I think we need more tests for fast append:
* Where no snapshots or branches exist
* Where the named branch does not exist, but main does
* Possibly others?

Reply:
Since the signature of the method now changes, the end user is required to pass SnapshotRef, are these checks required?

Reply:
@rdblue Added checks for above